ui/plugin-system/src/remote/remotePluginLoader.ts

```diff
@@ -41,10 +41,51 @@ const isPluginModuleResource = (pluginModule: unknown): pluginModule is PluginMo
   );
 };
 
-export const remotePluginLoader = (baseURL?: string): PluginLoader => {
+type RemotePluginLoaderOptions = {
+  /**
+   * The API path for fetching available Perses plugins. Used to construct the full URL to the `/api/v1/plugins` endpoint.
+   * @default ''
+   **/
+  apiPrefix?: string;
+  /**
+   * The base URL for loading plugin assets (e.g., JavaScript files). Used to construct the full URL to the `/plugins` directory
+   * @default ''
+   **/
+  baseUrl?: string;
+};
+
+type ParsedPluginOptions = {
+  pluginsApiPath: string;
+  pluginsAssetsPath: string;
+};
+
+const DEFAULT_PLUGINS_API_PATH = '/api/v1/plugins';
+const DEFAULT_PLUGINS_ASSETS_PATH = '/plugins';
+
+const paramToOptions = (options?: RemotePluginLoaderOptions): ParsedPluginOptions => {
+  if (options === undefined) {
+    return {
+      pluginsApiPath: DEFAULT_PLUGINS_API_PATH,
+      pluginsAssetsPath: DEFAULT_PLUGINS_ASSETS_PATH,
+    };
+  }
+
+  return {
+    pluginsApiPath: `${options?.apiPrefix ?? ''}${DEFAULT_PLUGINS_API_PATH}`,
```

Comment:
@jgbernalp should this also attempt to use the `baseUrl` if no `apiPrefix` is provided?

```suggestion
    pluginsApiPath: `${options?.apiPrefix ?? options?.baseUrl ?? ''}${DEFAULT_PLUGINS_API_PATH}`,
```

I think I prefer the explicit behavior but it does mean that the usage looks a little more verbose (not necessarily an issue), for example here:

https://github.com/perses/perses/pull/3377/files#diff-f4e96f71ee886ea641bc5d25c08b85739a3f85a93a45a3d3820c95d452781adaR87

becomes:
```diff
- <PluginRegistry pluginLoader={remotePluginLoader({ baseUrl: getBasePathName() })}>
+ <PluginRegistry pluginLoader={remotePluginLoader({ baseUrl: getBasePathName(), apiPrefix: getBasePathName() })}>
```

Reply:
I don't think so. The `baseUrl` and `apiPrefix` can be different, so the plugin loader should be soemething like:
```
const pluginLoader = remotePluginLoader({ baseUrl: getBasePathName(), apiPrefix: getApiPrefix() });
```

we don't have the `getApiPrefix` function yet, but these values should come from the [perses config](https://perses.dev/perses/docs/configuration/configuration/#:~:text=path%20%3Capi_prefix%3E/api-,api_prefix,-%3A%20%3Cstring%3E) using the [ConfigProvider](https://github.com/perses/perses/blob/main/ui/app/src/context/Config.tsx)